### PR TITLE
Update `CITATION.cff` to use `preferred-citation`

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
* Use `preferred-citation` to point to the FEniCS-preCICE paper
* Provide citation information about software on top level.
  * Open for discussion: Might be extended with other/future contributors that did not author the paper. Important: We need a proper and simple policy here. See also https://github.com/precice/fenics-adapter/graphs/contributors
  * Add a DOI? We could point to the preCICE distribution, but this is actually a superset of FEniCS-preCICE and the policy w.r.t. authorship is also independent. I would rather keep things separate and not provide a DOI. If we really want a DOI, we can still release on zenodo. This looks to me like the go-to-solution suggested by the CFF developers: https://citation-file-format.github.io/

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@richahert

File: CITATION.cff [hunk 1]
```diff
@@ -1,45 +1,71 @@
...
-  -
-    affiliation: "Technical University of Munich"
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: FEniCS-preCICE
+message: >-
+  If you use this software, please cite it using the
+  metadata from this file.
```
Comment (BenjaminRodenberg) [inline CITATION.cff:8]: We can also use this message for providing some further guidance. E.g. something similar to https://github.com/precice/.github/blob/9c7649853e4ca65c9961e655e8a808848efe6163/profile/README.md?plain=1#L15

**Review by ajaust — COMMENTED** [review]
Thanks for updating the citation information.

Should the discussion about the DOI and future authors be in separate issues?


When going through the repository I found another point that should potentially be discussed: Where/How do we want to maintain the citation information of the FEniCS adapter.

Currently, we have to maintain three places:
1. This `CITATION.cff` file. 

   The file and citation mechanism might be new to many people so we might need to explicitly mention GitHub's citation mechanism somewhere.
2. There is a citation section in the [README](https://github.com/precice/fenics-adapter/blob/develop/README.md).
3. There is a section on the [adapter website](https://precice.org/adapter-fenics.html#how-to-cite) that is part of the project.

File: CITATION.cff [hunk 2]
```diff
@@ -1,45 +1,71 @@
...
+  preCICE-adapter for the open source computing platform
+  FEniCS.
+license: LGPL-3.0
+commit: ' 9aa3e22'
+version: 1.4.0
+date-released: '2022-09-22'
+preferred-citation:
+  title: "FEniCS-preCICE: Coupling FEniCS to other Simulation Software"
+  type: "article"
+  abstract: "The new software FEniCS-preCICE is a middle software layer, sitting in between the existing finite-element library FEniCS and the coupling library preCICE. The middle layer simplifies coupling (existing) FEniCS application codes to other simulation software via preCICE. To this end, FEniCS-preCICE converts between FEniCS and preCICE mesh and data structures, provides easy-to-use coupling conditions, and manages data checkpointing for implicit coupling. The new software is a library itself and follows a FEniCS-native style. Only a few lines of additional code are necessary to prepare a FEniCS application code for coupling. We illustrate the functionality of FEniCS-preCICE by two examples: a FEniCS heat conduction code coupled to OpenFOAM and a FEniCS linear elasticity code coupled to SU2. The results of both scenarios are compared with other simulation software showing good agreement."
```
[the remark this answers is not in the document]
Comment (IshaanDesai) [reply CITATION.cff:41]: I would tend towards removing the abstract here; to make the BibTeX compact.

**Review by IshaanDesai — APPROVED** [review]

**Review by uekerman — APPROVED** [review]

**Review by ajaust — APPROVED** [review]

File: CITATION.cff (hunk 1 above)
Comment (BenjaminRodenberg) [reply CITATION.cff]: I added the following: "When using or referring to preCICE in academic publications, please follow the [citation guidelines](https://precice.org/fundamentals-literature-guide.html)."

File: CITATION.cff [hunk 3]
```diff
@@ -1,45 +1,71 @@
...
+    -
+      affiliation: "University of Stuttgart"
+      family-names: Uekermann
+      given-names: Benjamin
+      orcid: "https://orcid.org/0000-0002-1314-9969"
+  doi: 10.1016/j.softx.2021.100807
+  journal: "SoftwareX"
+  volume: 16
+  pages: 100807
+  year: 2021
```
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply CITATION.cff]: Added it :+1:

**Pull request merged**